src/utils/hooks/use-previous.tsx

```diff
@@ -0,0 +1,11 @@
+import { useEffect, useRef } from "react";
```

Comment:
this should be a `.ts` file instead of `.tsx` because it does not contain any HTML elements